### PR TITLE
Add hpmv! to BLAS in stdlib/LinearAlgebra

### PR DESCRIPTION
**Pull request by iolaszlo**

### CODE REVIEW

File: stdlib/LinearAlgebra/src/blas.jl
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/blas.jl]: remove one "a"

File: stdlib/LinearAlgebra/src/blas.jl
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/blas.jl]: described IN the reference

File: stdlib/LinearAlgebra/src/blas.jl
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/blas.jl]: "Hermitian matrix"

**Review by dkarrasch — CHANGES_REQUESTED** [review]
Just a couple of typos, otherwise LGTM.

File: stdlib/LinearAlgebra/src/blas.jl
Comment (iolaszlo) [reply stdlib/LinearAlgebra/src/blas.jl]: Removed  one "a".

File: stdlib/LinearAlgebra/src/blas.jl
Comment (iolaszlo) [reply stdlib/LinearAlgebra/src/blas.jl]: Added "in".

File: stdlib/LinearAlgebra/src/blas.jl
Comment (iolaszlo) [reply stdlib/LinearAlgebra/src/blas.jl]: Changed to "Hermitian matrix".

File: stdlib/LinearAlgebra/src/blas.jl
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/blas.jl]: I'm not sure but it seems to me that it should be `{T<:BlasComplex}`? `AP`, `x` and `y` are not converted in this method. Or are they converted somewhere implicitly? Also, the type parameters (after `where`) should be wrapped by curly brackets.

File: stdlib/LinearAlgebra/src/blas.jl
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/blas.jl]: I think we should mention here that `AP`, `x` and `y` must be complex.

File: stdlib/LinearAlgebra/test/blas.jl
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/blas.jl]: Nitpicking, but maybe you should stick a little closer to the style in this test file. The comment line is probably obsolete, and the testset should be set up only if `elty` is complex, similar to `her2k!` in line 26/27:
```julia
if elty in (ComplexF32, ComplexF64)
    @testset "hpmv!" begin
```
Finally, your tests are inside the `let`-block started on line 55/56, within which `n = 10` is used for vector lengths/matrix sizes. You could replace your loop over `N in 2:5` by a single test with `n = 10` instead of `N in 2:5`, or move your tests outside of the block.

File: stdlib/LinearAlgebra/test/blas.jl
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/blas.jl]: There is a complex random number generator, so `α = rand(elty)` is sufficient to have complex scalar numbers and achieve the same goal that you trying to achieve here. Same with the matrices below:
```julia
M = rand(elty, N, N) + elty(im)*rand(elty, N ,N)
```
generates two complex(!) `N` by `N` matrices, multiplies the second with `im` and then adds the two.

**Review by dkarrasch — CHANGES_REQUESTED** [review]

File: stdlib/LinearAlgebra/src/blas.jl
Comment (iolaszlo) [reply stdlib/LinearAlgebra/src/blas.jl]: Thanks for pointing this out. I agree also that  ```T```  should be```T <: BlasComplex``` and will apply this change  in order to enforce the correct input type. I haven't added any explicit type conversions for arrays and there are no implicit type conversions during the ```ccall``` as its  array inputs are of type Ptr{eltype} whose eltype can only be ComplexF32 and ComplexF64. Thus for example for real ```Float64``` array inputs we get a method error as expected:
```
 MethodError: no method matching Ptr{Array{Float64,2}}(::Array{Float64,1})
```

File: stdlib/LinearAlgebra/src/blas.jl
Comment (iolaszlo) [reply stdlib/LinearAlgebra/src/blas.jl]: Added the specs for the input types in the documentation.

File: stdlib/LinearAlgebra/test/blas.jl
Comment (iolaszlo) [reply stdlib/LinearAlgebra/test/blas.jl]: Moved the ```if``` statement outside of the ```@testset```. Use predefined ```n``` as the matrix dimension.

File: stdlib/LinearAlgebra/test/blas.jl
Comment (iolaszlo) [reply stdlib/LinearAlgebra/test/blas.jl]: Corrected the unnecessarily complicated input/output generation.

**Review by dkarrasch — APPROVED** [review]
Neat. Let's try to get another review before merging it.

**Comment by iolaszlo** [comment]
> Neat. Let's try to get another review before merging it.

Thank you for taking the time to review my commits.

**Comment by iolaszlo** [comment]
Could someone please retrigger the win64 and win32 builds, as their failure doesn't seem to be related to the current change. Thank you.

**Comment by ViralBShah** [comment]
Retriggered win64. Win32 had been a bit unhappy lately.

**Comment by iolaszlo** [comment]
> Retriggered win64. Win32 has been a bit unhappy lately.

Thank you, the win64 build succeeded. Win32 still seems unhappy as it gives a timeout currently. :)

**Comment by dkarrasch** [comment]
@iolaszlo In case you're not yet familiar with the conventions here, the "needs news" label means that you should write an entry in `NEWS.md` and announce the new feature in the `LinearAlgebra` section.

**Comment by iolaszlo** [comment]
> @iolaszlo In case you're not yet familiar with the conventions here, the "needs news" label means that you should write an entry in `NEWS.md` and announce the new feature in the `LinearAlgebra` section.

Thank you for the information. I just added the update in `NEWS.md` and rebased the commit onto the most current master branch.

**Comment by JeffBezanson** [comment]
Thanks, great first PR!

**Comment by StefanKarpinski** [comment]
Anyone else to review? @andreasnoack or @jiahao care to take a look?

**Comment by ViralBShah** [comment]
This is good to merge.

**Comment by iolaszlo** [comment]
Thank you guys for the feedback and the quick responses.

**Pull request merged**
